src/components/application_manager/test/commands/mobile/set_global_properties_request_test.cc

```diff
@@ -564,14 +604,179 @@ TEST_F(SetGlobalPropertiesRequestTest, Run_TTSIncorrectSyntax_Canceled) {
+  MessageSharedPtr msg = CreateMsgParams();
+  SmartObject help_prompt(smart_objects::SmartType_Array);
+  help_prompt[0][strings::text] = "help_prompt\\n";
+  (*msg)[strings::msg_params][strings::help_prompt] = help_prompt;
+
+  EXPECT_CALL(mock_app_manager_, application(kConnectionKey))
+      .WillOnce(Return(mock_app_));
+  EXPECT_CALL(mock_message_helper_, VerifyImageVrHelpItems(_, _, _)).Times(0);
+  EXPECT_CALL(mock_app_manager_, RemoveAppFromTTSGlobalPropertiesList(_))
+      .Times(0);
+
   EmptyExpectationsSetupHelper();
 
   SharedPtr<SetGlobalPropertiesRequest> command(
       CreateCommand<SetGlobalPropertiesRequest>(msg));
 
+  ExpectInvalidData();
+
+  command->Run();
+}
+
+TEST_F(SetGlobalPropertiesRequestTest, Run_InvalidVrHelpText_INVALID_DATA) {
+  MessageSharedPtr msg = CreateMsgParams();
+  SmartObject vr_help(smart_objects::SmartType_Array);
+  vr_help[0][strings::text] = "vr_help\\n";
+  (*msg)[strings::msg_params][strings::vr_help] = vr_help;
+
+  EXPECT_CALL(mock_app_manager_, application(kConnectionKey))
+      .WillOnce(Return(mock_app_));
+  EXPECT_CALL(mock_message_helper_, VerifyImageVrHelpItems(_, _, _))
+      .WillOnce(Return(mobile_apis::Result::SUCCESS));
+  EXPECT_CALL(mock_app_manager_, RemoveAppFromTTSGlobalPropertiesList(_))
+      .Times(0);
+
+  EmptyExpectationsSetupHelper();
+
+  SharedPtr<SetGlobalPropertiesRequest> command(
+      CreateCommand<SetGlobalPropertiesRequest>(msg));
+
+  ExpectInvalidData();
+
+  command->Run();
+}
+
+TEST_F(SetGlobalPropertiesRequestTest, Run_InvalidImageValue_INVALID_DATA) {
+  MessageSharedPtr msg = CreateMsgParams();
+  SmartObject vr_help(smart_objects::SmartType_Array);
+  vr_help[0][strings::text] = "vr_help";
+  vr_help[0][strings::image][strings::value] = "value\\n";
+  (*msg)[strings::msg_params][strings::vr_help] = vr_help;
+
+  EXPECT_CALL(mock_app_manager_, application(kConnectionKey))
+      .WillOnce(Return(mock_app_));
+  EXPECT_CALL(mock_message_helper_, VerifyImageVrHelpItems(_, _, _))
+      .WillOnce(Return(mobile_apis::Result::SUCCESS));
+  EXPECT_CALL(mock_app_manager_, RemoveAppFromTTSGlobalPropertiesList(_))
+      .Times(0);
+
+  EmptyExpectationsSetupHelper();
+
+  SharedPtr<SetGlobalPropertiesRequest> command(
+      CreateCommand<SetGlobalPropertiesRequest>(msg));
+
+  ExpectInvalidData();
+
+  command->Run();
+}
+
+TEST_F(SetGlobalPropertiesRequestTest, Run_InvalidMenuIcon_INVALID_DATA) {
+  MessageSharedPtr msg = CreateMsgParams();
+  SmartObject menu_icon(smart_objects::SmartType_Array);
+  menu_icon[strings::value] = "meun_icon\\n";
+  (*msg)[strings::msg_params][strings::menu_icon] = menu_icon;
+
+  EXPECT_CALL(mock_app_manager_, application(kConnectionKey))
+      .WillOnce(Return(mock_app_));
+  EXPECT_CALL(mock_message_helper_, VerifyImage(_, _, _))
+      .WillOnce(Return(mobile_apis::Result::SUCCESS));
+  EXPECT_CALL(mock_app_manager_, RemoveAppFromTTSGlobalPropertiesList(_))
+      .Times(0);
```

Comment:
@okozlovlux These three expectations repeat in several tests. Can you move them to a function?